### ISSUE TITLE
Orion browser: Error E036 in Macos and iOS

### ISSUE THREAD

**Issue by octaviz** [floccusaddon/floccus#1643]
### Which version of floccus are you using?

5.2

### How many bookmarks do you have, roughly?

400

### Are you using other means to sync bookmarks in parallel to floccus?

Firefox via Floccus and Mozilla Sync

### Sync method

WebDAV

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

iOS 17 and Macos 14

### What kind of WebDAV server are you using? (if relevant)

NAS Synology WebDAV server, in Firefox Bowser floccus working well

### Describe the Bug

"E036: Missing permissions to access the sync server Última sincronización: hace 2 semanas"

In Orion iOS/MacOS show this window.
<img width="495" alt="Xnip2024-06-12_18-03-40" src="https://github.com/floccusaddon/floccus/assets/856579/bbae776e-f86c-49fa-b268-77ea4c872431">

LOG: 
2024-06-12T15:55:16.665Z Starting sync process for account octavi@XXXXX.homeftp.org
2024-06-12T15:55:16.711Z onSyncStart: begin
2024-06-12T15:55:16.712Z onSyncFail
2024-06-12T15:55:16.712Z Syncing failed with E036: Missing permissions to access the sync server
2024-06-12T15:55:16.713Z onSyncFail

This error is for when you use the Firefox extension, when you use the Chrome extension you don't get any error and it doesn't synchronize or do anything, it even shows the "Todo Bien" message in green color, but it doesn't really do anything.

### Expected Behavior

That it did the synchronization without errors, since the WebDAV server works in other browsers with floccus.

### To Reproduce

I have tried several options and all give me the same result, I import the profile I have with all the settings and when it tries to synchronize it gives the error

### Debug log provided

- [X] I have provided a debug log file

**Comment by marcelklehr** [comment]
Hello @octaviz 
Make sure to disable Firefox Sync when using Floccus, it can cause data corruption. For the missing permissions, can you check the addon settings in about:addons and check whether floccus has network access?

**Comment by marcelklehr** [comment]
> when you use the Chrome extension you don't get any error and it doesn't synchronize or do anything, it even shows the "Todo Bien" message in green color, but it doesn't really do anything.

Could it be that the folder you have selected to be synchronized is empty? Then floccus will not do anything.

**Comment by octaviz** [comment]
I don't think I mentioned that I am using it on the new Orion Browser, you can add FF and Chrome extensions. Maybe that's why it doesn't work, it's an Orion thing I think.

I just created from scratch in my Firefbox a new profile with the same configuration that I put in Orion and it works. Orion is a good browser with lots of options, I don't know if you know it. 

https://kagi.com/orion/

**Comment by marcelklehr** [comment]
Glad it's working now.

**Comment by octaviz** [comment]
Work in FireFox, in Orion Browser nop

**Comment by marcelklehr** [comment]
Can you try export the profile you created in Firefox and import in floccus in Orion?

**Comment by octaviz** [comment]
If that is what I have done, it gives error E036, I have also created it manually and the same error. 

I think the problem is with the Orion browser and the permissions or some access, because many extensions also do not work as they should.

**Comment by fufu65** [comment]
Hi,
on my Windows 10 PC I have Floccus installed on 4 browsers: Firefox, Edge, Chrome and Brave; two distinct NextCloud accounts configured on all browsers.
Version 5.1.4 worked fine; today's update to version 5.2.0 has problems with some browsers:
- Firefox with 5.2.0 is ok;
- Chrome and Brave with 5.2.0 has problems: "E017: Network error: Check your network connection and your account details Last synchronized: 6 ore ago"
- Edge with 5.1.4 is ok, but not yet updated...

**Comment by marcelklehr** [comment]
Hi @fufu65 could you please open a new issue? <3

**Comment by strich3** [comment]
Still doesn't work yet? Having the same issues

**Comment by marcelklehr** [comment]
So far, I'm assuming this is a bug in Orion browser :/

**Comment by strich3** [comment]
Well, I love Orion but I really need to well sync my bookmarks. But I totaly understand that it is not your responsibility to get it work :D

**Comment by mdl054** [comment]
I can report the same issue on my end

**Comment by scgf** [comment]
Orion is very picky about which extensions it works with. It doesn't like xbrowsersync either. It does work with Bookmarkhub though. This extension needs to be run manually on each browser to sync bookmarks - which it does very quickly.

https://github.com/dudor/BookmarkHub

**Comment by marcelklehr** [comment]
I've published a possible fix for this as part of v5.3.1

**Comment by strich3** [comment]
I could try it if I would know how to install the addon :)

**Comment by marcelklehr** [comment]
I would assume you can go to Chrome Webstore and press install?

**Comment by strich3** [comment]
It isn't merged yet, or is it?

**Comment by marcelklehr** [comment]
It is :)

> I've published a possible fix for this as part of v5.3.1

**Comment by strich3** [comment]
Still same issue and same error, installed the firefox addon because the chrome addon doesn‘t work at all

**Comment by egigoka** [comment]
Nextcloud Bookmarks gives this error, but Linkwarden syncs normally for me!

**Comment by marcelklehr** [comment]
mmmh

**Comment by kotlasu** [comment]
Dang, I wanted to move away from Firefox (and Firefox Sync) to a combination of Floccus + Orion Browser (on iOS) - thanks to this thread I know that GIT sync is giving me a headache (dreaded E036). It's time to try syncing via Linkwarden then.

**Comment by marcelklehr** [comment]
Mh, something is still off with Orion's permissions API, but I've simply added an exception now for Orion, which makes floccus work with Orion 🎉

**Comment by scgf** [comment]
I just tried floccus and it's still not working for me - how can I achieve what you've achieved? Do i need to add and exception somewhere? Thank you.

**Comment by marcelklehr** [comment]
heh, I'm not that fast. I still have to release the fix

**Comment by scgf** [comment]
Sorry :)

**Comment by scgf** [comment]
I’ve installed Floccus v5.4.5 and it still doesn’t work. I’ve tried Nextcloud bookmarks, two different WebDAV addresses and Google. When I have it configured and try to get it to sync it doesn’t even try. Absolutely nothing happens and it doesn’t even generate any logs.  I’m using Orion 0.99.132-beta. I used the Floccus xpi version.

**Comment by marcelklehr** [comment]
Mh, I can confirm in the released version it doesn't work.

**Comment by marcelklehr** [comment]
It seems their permissions system is not simply not implemented but consistently returns false for all websites, and I see no way to identify that we're running on Orion either, sigh. This seems to be the revent topic on orionfeedback : https://orionfeedback.org/d/637-res-permission-request-prompts-dont-show-up

**Comment by marcelklehr** [comment]
I found a way to identify Orion, now we can ignore the permission system reliably and make sync work on Orion 🎉

**Comment by marcelklehr** [comment]
v5.6.0 is out now with Orion support

**Comment by scgf** [comment]
I still doesn't work. Clicking on 'download once' does nothing, clicking on 'sync now' does nothing. It doesn't even show an error. I'm using Floccus 5.6 and Orion version 0.99.135.0.1-beta (WebKit 622.1.15.19.2). I'm using the same Floccus settings as I use on all my browsers - which work fine.

**Comment by marcelklehr** [comment]
Are you using the Chrome extension or the Firefox extension? Only the firefox extension works on Orion, I believe

**Comment by scgf** [comment]
I'm using the Firefox extension.

**Comment by sbugzu** [comment]
The Firefox extension doesn't work on my MacOS either, but the Chrome extension works fine.
After installing and configuring it, try restarting your browser.
The extension didn't work at first, but restarting it fixed the problem.

**Comment by scgf** [comment]
Yes, I can confirm the Chrome extension works perfectly on macOS. Thank you @sbugzu.
